src/snowcli/cli/snowpark/procedure_coverage/manager.py

```diff
@@ -0,0 +1,138 @@
+import logging
+import os
+import tempfile
+from enum import Enum
+from pathlib import Path
+
+import coverage
+import snowflake
+import typer
+from click import ClickException
+from snowflake.connector.cursor import SnowflakeCursor
+
+from snowcli import utils
+from snowcli.cli.common.sql_execution import SqlExecutionMixin
+from snowcli.cli.stage.manager import StageManager
+from snowcli.snow_connector import generate_signature_from_params
+from snowcli.utils import generate_deploy_stage_name
+
+log = logging.getLogger(__name__)
+
+
+class ReportOutputOptions(str, Enum):
+    html = "html"
+    json = "json"
+    lcov = "lcov"
+
+
+class UnknownOutputFormatError(ClickException):
+    def __init__(self, output_format: ReportOutputOptions):
+        super().__init__(f"Unknown output format '{output_format}'")
+
+
+class ProcedureCoverageManager(SqlExecutionMixin):
+    def report(
+        self,
+        name: str,
+        input_parameters: str,
+        output_format: ReportOutputOptions,
+        store_as_comment: bool,
+    ) -> str:
+        conn = self._conn
+        deploy_dict = utils.get_deploy_names(
+            conn.ctx.database,
+            conn.ctx.schema,
+            generate_deploy_stage_name(
+                name,
+                input_parameters,
+            ),
+        )
```

Comment:
Have you considered to moving this to manger instead of keeping this a separate util?